README.md

```diff
@@ -279,6 +279,31 @@ role.
 Similar to `controller` and `vlan`, the `parent` references the connection profile in
 the ansible role.
 
+#### `type: infiniband`
+
+For the infiniband connection, the following options are supported:
+
+- `p_key`: The infiniband P_Key to use for the device. The default value of `-1` means
```

Comment:
In the other PR it seems that `-1` is not intended to be set by the user but just an internal default value. What is correct? Also, it seemed to me that it is mandatory to set this, is it not? This is confusing.

Reply:
https://www.kernel.org/doc/Documentation/infiniband/ipoib.txt, the kernel did not mention `-1` as the internal default value

But NM uses `-1` means to use the  default P_Key (aka "the P_Key at index 0")
```
=== [p-key] ===
[NM property description]
The InfiniBand P_Key to use for this device. A value of -1 means to use the default P_Key (aka "the P_Key at index 0"). Otherwise, it is a 16-bit unsigned integer, whose high bit is set if it is a "full membership" P_Key.
```

Reply:
Yes, it is mandatory to set, if user does not set the pkey explicitly, the default value `-1` will be choose for the role which  corresponds to the default P_Key in kernel (aka "the P_Key at index 0")

Reply:
maybe I should change the `unset` into `not explicitly set`

Reply:
Well, then pkey is not mandatory, if the user does not set the pkey explicitly (to a value different than -1), then the user creating the connection on the physical infiniband interface (not ipoib mode), in such situation,  https://github.com/linux-system-roles/network/pull/495/files#diff-6b3f5f6149ac1330c5e2c832286b63b3597158a998f2244ddbec2045ace1f794R2022 will raise an error when missing the physical infiniband interface.

Reply:
> Well, then pkey is not mandatory, if the user does not set the pkey explicitly (to a value different than -1), then the user creating the connection on the physical infiniband interface (not ipoib mode)

Ok, so if the `p_key` setting also controls the infiniband mode (physical vs. ipoib), then this should be mentioned in the documentation. Also, the documentation mentions that it would use the P_Key at index 0, so what happens when  p_key is set to 0 by the user? Based on the descriptions so far it seems that this also implies physical mode, since p_key -1 means 0 means physical mode? The code does not check for p_key == 0, though.

Reply:
Well, if the user set the pkey to 0, then ipoib will be created. I referred from "There are special requirements for the names of IPoIB P_Key interfaces. All IPoIB P_Keys range from 0x0000 to 0x7fff, and the high bit, 0x8000, denotes that membership in a P_Key is full membership instead of partial membership." [1] 

For "use the P_Key at index 0", I would interprete as using the base infiniband fabric, and for the pkey value greater or equal to zero,  I would interprete as creating the partition or virtual subnet.  

1. https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-configuring_ipoib

Reply:
README updated.